sdk/python/lib/pulumi/runtime/resource.py

```diff
@@ -644,14 +644,14 @@ def __init__(self,
 def convert_providers(
         provider: Optional['ProviderResource'],
         providers: Optional[Union[Mapping[str, 'ProviderResource'],
-                                  List['ProviderResource']]]) -> Mapping[str, 'ProviderResource']:
+                                  Iterable['ProviderResource']]]) -> Mapping[str, 'ProviderResource']:
```

Comment:
Is there a reason you used `Iterable` here instead of `Sequence`? Isn't `Mapping` itself a subclass of `Iterable`? Does that mean this is just `Optional[Iterable['ProviderResource']]`?

Reply:
Oops, no, just an oversight. I can change this to `Sequence` if you'd prefer, but all `Sequence[T]`s implement `Iterable[T]` so it's not too much different.

`Mapping` *is* a subclass of `Iterable`, but `Mapping[str, ProviderResource]` impls `Iterable[[str, ProviderResource]]`. The idea here is that this function takes a list-like *or* a dict-like.

Reply:
Right right, makes sense.

Yeah, I think let's just change this to `Sequence` to keep things consistent. Otherwise, the change looks good to me.

Reply:
Sure thing, done!